pkg/controller/rmdworkload/rmdworkload_controller.go

```diff
@@ -143,40 +137,23 @@ func (r *ReconcileRmdWorkload) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	// Create a map of node state namespaces for RMD pod discovery.
-	// Nodestates and RMD pods are created with the same namespace by node_controller.
-	// This is the same namespace as the parent node, or default if not set.
-	rmdWorkloadName := rmdWorkload.GetObjectMeta().GetName()
-	nodeNamespaces := make(map[string]string)
-	for _, rmdNodeState := range rmdNodeStates.Items {
-		nodeNamespaces[rmdNodeState.Spec.Node] = rmdNodeState.GetObjectMeta().GetNamespace()
+	// Discover all RMD instances that the reconciled RmdWorkload is targeting.
+	// Add or Update those instances with the reconciled RmdWorkload accordingly
+	targetNodes, err := r.findTargetedNodes(request, rmdNodeStates, rmdWorkload)
+	if err != nil {
+		return reconcile.Result{}, err
 	}
-	// Loop through nodes listed in RmdWorkload Spec, add/update workloads where necessary.
-	for _, nodeName := range rmdWorkload.Spec.Nodes {
-		// Get node service address
-		address, err := r.getPodAddress(nodeName, nodeNamespaces[nodeName])
-		if err != nil {
-			reqLogger.Error(err, "Failed to get pod address")
-			return reconcile.Result{}, err
-		}
 
-		activeWorkloads, err := r.rmdClient.GetWorkloads(address)
-		if err != nil {
-			reqLogger.Info("Could not GET workloads.", "Error:", err)
-			return reconcile.Result{}, err
-		}
-
-		workload := rmd.FindWorkloadByName(activeWorkloads, rmdWorkloadName)
-		if workload.UUID == "" {
+	for _, node := range targetNodes {
```

Comment:
again, for consistency's sake, I would recommend `for _, targetedNode := range targetedNodes {`